doc/dev/sharp_bits.rst

```diff
@@ -0,0 +1,1015 @@
+Debugging and sharp bits
+========================
+
+Catalyst is designed to allow you to take the tools and code patterns you are
+familiar with when exploring quantum computing (such as Python, NumPy, JAX,
+and PennyLane), while unlocking faster execution and the ability to run
+hybrid quantum-classical workflows on accelerator devices.
+
+Similar to JAX, Catalyst does this via the :func:`@qjit <.qjit>` decorator, which captures
+hybrid programs written in Python, PennyLane, and JAX, and compiles them to
+native machine code --- preserving important aspects like conditional
```

Comment:
It might be nice to be more precise with what you mean by native machine code. The reader might think we support some QPUs.

Reply:
Good point, do you have a suggestion? 🤔